inductiva/fluids/scenarios/coastal_area/coastal_area.py

```diff
@@ -75,6 +77,58 @@ def from_text_file(
         depths = np.loadtxt(text_file_path)
         return cls(depths, x_range, y_range)
 
+    @classmethod
+    def from_random_depths(
+        cls,
+        n: int,
+        x_range,
+        y_range,
+        max_depth: float = 10,
+        initial_roughness: float = 1,
+        roughness_factor: float = 0.5,
+        percentile_above_water: float = 20,
+    ):
+        """Creates a `Bathymetry` object with random depths.
+
+        A grid of random depths of size (2^n + 1) x (2^n + 1) is created, where
+        n is an integer.
+
+        The depths are randomly generated as follows:
+        1. A maximum depth value `max_depth` is used to specify the range of
+        depths to assign to the corners of the grid. The corners on the lower x
+        boundary (East) are assumed to be below water (i.e. have 0 < depths <
+        `max_depth`). The corners on the upper x boundary (West) are assumed to
+        be above water (i.e. have - `max_depth` < depths < 0).
+        2. The depths of the remaining points are generated by applying the
+        Diamond-Square algorithm. The randomness of the algorithm is controlled
+        by the `initial_roughness` and `roughness_factor` parameters, that set
+        the initial range of randomness and the rate at which it decreases over
+        iterations of the algorithm, respectively.
+        
+        Args:
+            n: Sets the size of the grid of random depths, (2^n + 1) x (2^n + 
+              1).
+            x_range: The range of x values, in meters.
+            y_range: The range of y values, in meters.
+            max_depth: Maximum depth value, in meters.
+            initial_roughness: Initial roughness value, in meters. Controls the
+              initial range of randomness of the Diamond-Square algorithm.
+            roughness_factor: Roughness factor. Must be between 0 and 1.
+              Controls the rate at which the range of randomness of the
+              Diamond-Square algorithm decreases over iterations.
+            percentile_above_water: Percentile of the depths that must be above
+              water. Must be between 0 and 100.
+        """
+
+        depths = random_bathymetry_utils.create_random_bathymetry(
```

Comment:
This is ok for now, but maybe we could generalize a bit the code in random_bathymetry_utils and have it in inductiva/generative because we can use it on a number of other situations (e.g. @IvanPombo may want to use this too).

Reply:
I was actually thinking of using an open-source library, but given this is already here, I will adapt to my case and then we think about the generalization :)